app/services/decisions/client_decision_tree.rb

```diff
@@ -33,6 +33,14 @@ def after_has_partner
       end
     end
 
+    def after_details
```

Comment:
So it has effect if the following flow takes place:
1. Provider completes full journey for u18
2. Provider returns to client details and changes dob to result in age > 18
3. Provider navigates to tasklist and sees IOJ with Completed badge

Reply:
Yep, I think what you describe can be an issue. Is a bit theoretical as I can hardly see in real life a provider changing the DOB in such a substantial way the client goes from being under 18 to over 18, but yes, the code would allow it so is possible and is good to cover our bases.

But perhaps separating these passporting rules from the IOJ table is recommendable. Having it separate will let us nuke the Ioj record instead of having to worry about updating the types. Or maybe we don't even need to worry about deleting previous Ioj, if entered because the task list can take care of it (*).

Also, not sure if this will happen in the end, but there were discussions about "sampling" a % of passported cases, meaning even if some passporting rules are met, we still take the provider to the Ioj page. Like A/B testing. If we are reusing the `types` attribute for both, this could have problems or more complicated logic.

(*) Regarding the task list, it was anticipated this would happen. The task `app/presenters/tasks/ioj.rb` was not completed, awaiting the passporting rules. There is a comment in `#not_applicable?`. This method in theory could check if we've "activated" any of the passporting rules by looking at the attribute wherever we are writing them.

(**) Additionally the Check your answers will have to present a different "Section" even tho look very similar to our current Ioj, because it will not have "Change" link if I'm interpreting correctly the designs.

Happy to discuss this more in a chat! or with Carolina! I'm worried this might be a bit more complicated than perhaps we are thinking.